pail/s3_bucket.go

```diff
@@ -0,0 +1,551 @@
+package pail
+
+import (
+	"context"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/pkg/errors"
+)
+
+type s3BucketSmall struct {
+	s3Bucket
+}
+
+type s3BucketLarge struct {
+	s3Bucket
+}
+
+type s3Bucket struct {
+	name   string
+	prefix string
+	sess   *session.Session
+	svc    *s3.S3
+}
+
+type S3Options struct {
```

Comment:
It's a good practice on public structs and functions to write a docstring.

Reply:
Added.